### PR TITLE
[ASV-964-963-965] [ASV-990] MTP for App of the week

### PR DESCRIPTION
**Pull request by JDSM01**
**What does this PR do?**

 
  Adds the App of the Week (Editorial). (All the requests need to display the view are made in this ticket also)
  Color pick for the Game of the Week card
  Install card animation
  StatusBar transparent

**Database changed?**

   No

**Where should the reviewer start?**

EditorialFragment.java

**How should this be manually tested?**

Go to the view and check for layout errors. Monkey test with the view. Check if request are being done correctly

**What are the relevant tickets?**

https://aptoide.atlassian.net/browse/ASV-964
https://aptoide.atlassian.net/browse/ASV-841
https://aptoide.atlassian.net/browse/ASV-990
https://github.com/Aptoide/aptoide-client-v8/pull/691
https://github.com/Aptoide/aptoide-client-v8/pull/692

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Partners build
- [ ] Unit tests pass
- [ ] Functional QA tests pass

### CODE REVIEW

File: app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialItemsViewHolder.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          //TODO add video
+          video.setVisibility(View.VISIBLE);
+        }
+      }
+      EditorialMedia editorialMedia = editorialItem.getMedia()
+          .get(0);
+      if (editorialMedia.hasDescription()) {
+        description.setText(editorialMedia.getDescription());
+        description.setVisibility(View.VISIBLE);
+        //TODO add description here in the future
```
Comment (paquiquinho) [inline app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialItemsViewHolder.java]: is this TODO right here?

File: app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialPresenter.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ * Created by D01 on 27/08/2018.
+ */
+
+public class EditorialPresenter implements Presenter {
+
+  private final EditorialView view;
+  private final EditorialManager editorialManager;
+  private final Scheduler viewScheduler;
+  private final CrashReport crashReporter;
+  private final EditorialNavigator appOftheWeekNavigator;
```
Comment (paquiquinho) [inline app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialPresenter.java]: not used

File: app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialViewModel.java [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    this.fileSize = fileSize;
+    this.path = path;
+    this.backgroundImage = backgroundImage;
+    this.pathAlt = pathAlt;
+    this.md5 = md5;
+    this.loading = false;
+    this.error = null;
+  }
+
+  public EditorialViewModel() {
```
Comment (paquiquinho) [inline app/src/vanilla/java/cm/aptoide/pt/app/view/EditorialViewModel.java]: This constructor is never used, do we really need it?

File: dataprovider/src/main/java/cm/aptoide/pt/dataprovider/model/v7/EditorialCard.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+    public App getApp() {
+      return app;
+    }
+
+    public void setApp(App app) {
+      this.app = app;
+    }
+
+    public String getBackground_image() {
```
Comment (paquiquinho) [inline dataprovider/src/main/java/cm/aptoide/pt/dataprovider/model/v7/EditorialCard.java]: What's up with the _ both on attributes and methods here (and the one bellow)?

File: app/src/main/java/cm/aptoide/pt/home/HomeNavigator.java [hunk 5]
```diff
@@ -72,4 +74,14 @@ public void navigateToMyAccount() {
   public void navigateToAppCoinsInformationView() {
     fragmentNavigator.navigateTo(new AppCoinsInfoFragment(), true);
   }
+
+  public void navigateToEditorial() {
+    Bundle bundle = new Bundle();
+    //TODO change for the real values when merging
+    bundle.putString("cardId", "72635teguyegr");
+    bundle.putString("appName", "App Title");
```
Comment (paquiquinho) [inline app/src/main/java/cm/aptoide/pt/home/HomeNavigator.java]: Not needed anymore

File: app/src/main/java/cm/aptoide/pt/util/AppBarStateChangeListener.java [hunk 6]
```diff
@@ -36,6 +41,6 @@
   public abstract void onStateChanged(AppBarLayout appBarLayout, State state);
 
   public enum State {
-    EXPANDED, COLLAPSED, IDLE
+    EXPANDED, COLLAPSED, IDLE, COLLAPSED_COMPLETELY
```
[the remark this answers is not in the document]
Comment (JDSM01) [reply app/src/main/java/cm/aptoide/pt/util/AppBarStateChangeListener.java]: Went with MOVING

**Pull request merged**